CRM/Core/Form.php

```diff
@@ -420,6 +420,21 @@ public function &add(
    * This is a virtual function and should be redefined if needed.
    */
   public function preProcess() {
+    try {
+      $entity = $this->getDefaultEntity();
+      if (empty($this->_id)) {
+        return;
```

Comment:
this line shouldn't be in preProcess

Reply:
Haha! It was written as a quick proof of concept.  Intention was to check if $this->_id existed and not run the custom data stuff if it doesn't

Reply:
I'm just trying to think of any risks from security POV of this - haven't yet but need to read the code more. Also, I wonder if we should also check the entity supports custom data.

Reply:
Ok, is there an easy way to check if the entity supports custom data?

Reply:
CRM_Core_SelectValues::customGroupExtends()

Reply:
(might be worth adding an isSupported helper - would save a couple of lines in more than one place probably & be good for readability)